clients/lcdproc/disk.c

```diff
@@ -124,32 +137,71 @@ disk_screen(int rep, int display, int *flags_ptr)
 	sock_printf(sock, "widget_set D f 1 2 %i %i %i %i v 12\n", lcd_wid, lcd_hgt, lcd_wid, count);
 	for (i = 0; i < count; i++) {
 		char tmp[lcd_wid + 1];	/* should be large enough */
+		char cap[8] = {'\0'};
+		char dev[256];
+		u_int64_t size = 0;
+		int full = 0;
 
-		if (table[i].dev[0] == '\0')
+		// Eewww, goto... but shy of restructuring everything it's 
+		// the cleanest way to skip the rest of the outer loop. :-/
+		next_mnt:
+
+		if (mnt[i].mpoint[0] == '\0')
 			continue;
 
-		if (i >= num_disks) {	/* Make sure we have enough lines... */
-			sock_printf(sock, "widget_add D s%i string -in f\n", i);
-			sock_printf(sock, "widget_add D h%i hbar -in f\n", i);
+		// Prescreen for entries to ignore.
+		for (int ign_ct=0; ign_ct<ignore_count; ign_ct++) {
+			if (strcmp(mnt[i].mpoint, disk_ignore[ign_ct])==0) {
+				ig_offset++;
+				i++;
+				goto next_mnt; // Behold, the one legitimate use for goto ;-)
+			}
```

Comment:
There are many legitimate uses for goto, this is not one of them. Manually messing with loop counters is _exactly_, what makes programs hard to understand and why goto is so unpopular.

I suggest you define a helper function and use something like:
```
if (is_ignored(mnt[i].mpoint))
        continue;
```

Reply:
in this scenario does it make sense to condense the two ifs into one? I'm not sure what situations would cause a null mountpoint to appear in the list (except if you run past `count`... in which case we should already be out of the for loop.)

If it is a possibility then I think it makes sense to also increment ig_offset because otherwise you'd get a blank line in the disk screen.

Reply:
Yeah, sure. My code was just a template, nothing to be used directly.

But I do hope, the code can be simplified, by making `i_widget` a loop variable and getting rid of `if_offset` ...

(I hope I understood your question correctly.)

Reply:
If it helps clarify, this is what I was thinking of:
```
		if (mnt[i].mpoint[0] == '\0' || disk_is_ignored(mnt[i].mpoint, disk_ignore)) {
			ig_offset++;
			continue;
		}
```

For the other point - you are proposing we make the "widget" index the real loop counter and maintain the offset of "skipped" entries in `mnt[]` instead? Maybe I'm missing something but I don't see a "clean" way to maintain two indices that might diverge (because the ignore entries can appear at any position in `mnt[]`) without either explicitly storing them (or an offset) or getting into things like pointer arithmetic. 

My main question was whether it's ever possible to observe a null mnt.name in practice. But I guess it's moot since for our intent it doesn't matter if it's null or on the ignore list, either way it should not be displayed and we should move on to the next entry.

Reply:
I doubt I'm any more familiar with the code then you are. So I don't know either. But I think for the purpose of this PR you can assume (even wrongly), that the original code worked sensibly. So if a null mnt.name doesn't make sense, you a free to pretend, it can't happen. :-) If the original code is broken, we always can (and I actually prefer) to fix it in a separate commit.

In C you certainly can have two distinct loop variables. Something like:
```
for (i = 0, j = 0; i < 10; ++i, ++j)
        if (...)
                ++j;
```
is perfectly valid and actually quite common.

Reply:
Sure, I know you can have multiple loop variables but I don't think it adds any benefit  here because we don't want to auto-increment i_widget on continue; only if we make it through the end of the loop. Unless you're proposing the logic be completely inverted so that the primary loop controller is the widget index and the index into `mnt[]` is ancillary and bumped on continue.

Reply:
There are several possible patterns. One is to swap primary and secondary loop variable as you propose. (Probably the most intuitive.) An other is to not auto increment the second variable, but increment it at the end of the loop manually. The worst (but probably still better then having an extra variable) would be to decrement before the continue statement. ;-)

I know, this really is nit picking. However at this time lcdproc is entirely maintained and improved by people who are very much not familiar with the code and just stop by to scratch some itch. Thus I feel it is important, that the code be as readable and comprehensible as possible.

Reply:
Please have a look at #182. I've eliminated the use of `ig_offset` and the sole additional counter is now `i_widget`;  Unfortunately there is no way to make it entirely scoped to just the loop because its value is required afterwards to remove extraneous line entries and update the static count. (This could probably be integrated into the main FOR as well, but at that point the nesting of control structures starts getting a little onerous. 

The change I have made there sounds like it aligns with the second option you describe above as best as possible. 

I certainly don't take any offense at your feedback here; I would likely do the same for any of my projects that were in "maintenance" mode in order to ensure I would be able to pick things up  easily again if necessary. (in fact, that's why I'm in favor of leaving the loop index alone as much as possible; it makes it easier for someone that has a prior familiarity to pick things up again without having to re-learn some things they might still recall. )

Reply:
Yes, I'm perfectly happy with the way you have handled this in #182. 

In the case of lcdproc there is little hope, that people, who know the code, come back. It's more about making it easy to comprehend for new people.